lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -64,8 +60,10 @@ def install!
               app_native_targets = add_app_targets
               app_resource_bundle_targets = add_resources_bundle_targets(app_file_accessors)
 
-              add_files_to_build_phases(native_target, test_native_targets, app_native_targets)
-              validate_targets_contain_sources(test_native_targets + app_native_targets.values + [native_target])
+              if target.should_build?
+                add_files_to_build_phases(native_target, test_native_targets, app_native_targets)
```

Comment:
oops this is wrong. This does not add the source files of test targets and app targets. Will need to fix.

Reply:
fixed, ready now.

Reply:
also updated tests to catch this.